### PR TITLE
[eigen3] Upgrade to 3.4.0

### PR DESCRIPTION
**Pull request by tusharb86**
**Describe the pull request**

- #### What does your PR fix?  
  Updates [eigen3](https://gitlab.com/libeigen/eigen/) to version 3.4.0.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/eigen3/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,16 @@
 {
   "name": "eigen3",
-  "version-string": "3.3.9",
-  "port-version": 1,
+  "version-string": "3.4.0",
   "description": "C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.",
-  "homepage": "http://eigen.tuxfamily.org"
+  "homepage": "http://eigen.tuxfamily.org",
```
Comment (dg0yt) [inline ports/eigen3/vcpkg.json:5]: No https homepage?

**Review by dg0yt — COMMENTED** [review]
There is already #19665.

File: ports/eigen3/vcpkg.json (hunk 1 above)
Comment (ras0219-msft) [reply ports/eigen3/vcpkg.json:5]: Fixed in #19665

**Comment by ras0219-msft** [comment]
Duplicate of #19665 -- I'm going to see about pushing that one to completion.

**Pull request closed without merging**